aptos-move/framework/aptos-framework/sources/aggregator/optional_aggregator.move

```diff
@@ -79,7 +79,7 @@ module aptos_framework::optional_aggregator {
     }
 
     /// Switches between parallelizable and non-parallelizable implementations.
-    public(friend) fun switch(optional_aggregator: &mut OptionalAggregator) {
+    public fun switch(optional_aggregator: &mut OptionalAggregator) {
```

Comment:
@movekevin  Making this public would allow anyone with a coin to upgrade the supply? e.g.
```
let maybe_supply = &borrow_global<CoinInfo<CoinType>>(coin_address<CoinType>()).supply;
// assume we have option::is_some(maybe_supply))
let supply = option::borrow(maybe_supply);
optional_aggregator::switch(supply)
```

Reply:
I don't think that's possible as the coin module currently does not return that supply object (OptionalAggregator) anywhere. The code you wrote there is only possible if it's done/exposed by a function in coin module. Another module cannot just read supply off CoinInfo like that.

Reply:
This optional_aggregator module already has a friend-only restriction on creation so I think we don't need the same restriction on switch().

Reply:
The only 2 entries currently to switching are:
1. Calling switch directly. But that's not possible without access to a mutable ref to OptionalAggregator, which can only be created by calling a friend-only function.
2. Calling coin::upgrade_supply, which is guarded by a governance-controlled config.